lib/src/kbasesearchengine/parse/KeywordParser.java

```diff
@@ -268,7 +270,14 @@ private static Object transform(
             } catch (IllegalArgumentException e) {
                 throw new ObjectParseException(e.getMessage(), e);
             }
-            Set<GUID> guids = lookup.resolveRefs(objectRefPath, unresolvedGUIDs);
+            Set<ResolvedReference> resrefs = lookup.resolveRefs(objectRefPath, unresolvedGUIDs);
```

Comment:
lookup.resolveRefs returns a Set<ResolvedReference> which is then being converted to a Set<GUID>. This change was made to use cached resolvedRefs. 

https://github.com/arfathpasha/KBaseSearchEngine/commit/fd71a20e451f9b9fc0bafb10191f384f9616b04e#diff-14fdf7569af870a20d0ae2cc960997c6

Reply:
Sorry, I'm still confused here. resolveRefs() was changed to return Set<ResolvedReference> rather than Set<GUID>. But this is the only place in the codebase where the method is called, and it immediately changes the Set<ResolvedReference> back to Set<GUID> and never uses the Set<ResolvedReference> at all. 

It seems to me that there's no reason for this change, and I don't understand how caching the ResolvedReferences has anything to do with it - that happens regardless of the return type right?

Reply:
Maybe I should clarify that 'this change' = the return type of the method, not necessarily changes to the method implementation itself.

Reply:
resolveRefs() is used in three places - 

https://github.com/arfathpasha/KBaseSearchEngine/blob/dev-recursive-indexing/lib/src/kbasesearchengine/main/IndexerWorker.java#L831

https://github.com/arfathpasha/KBaseSearchEngine/blob/dev-recursive-indexing/lib/src/kbasesearchengine/main/IndexerWorker.java#L964

https://github.com/arfathpasha/KBaseSearchEngine/blob/dev-recursive-indexing/lib/src/kbasesearchengine/parse/KeywordParser.java#L273

I have not analyzed to see if they all cascade down to a single use since there is caching involved and this should compensate for any inefficiency. I think my overall goal here was to make sure I resolve a guid before I use it simply as a safety measure.

Reply:
Doh, when I searched the diff IndexerWorker hadn't been loaded by github so those uses didn't show up.